lazyflow/operators/opRelabelConsecutive.py

```diff
@@ -1,26 +1,506 @@
+###############################################################################
+#   ilastik: interactive learning and segmentation toolkit
+#
+#       Copyright (C) 2011-2024, the ilastik developers
+#                                <team@ilastik.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# In addition, as a special exception, the copyright holders of
+# ilastik give you permission to combine ilastik with applets,
+# workflows and plugins which are not covered under the GNU
+# General Public License.
+#
+# See the LICENSE file for details. License information is also available
+# on the ilastik web site at:
+#          http://ilastik.org/license.html
+###############################################################################
+import collections
 import logging
+import time
+from functools import partial
+from typing import Dict, Tuple, Union
+
+import numpy
+import numpy.typing as npt
 import vigra
-from lazyflow.graph import Operator, InputSlot, OutputSlot
-from lazyflow.utility import timeLogged
+
+from lazyflow.graph import InputSlot, Operator, OutputSlot
+from lazyflow.operators.generic import OpPixelOperator
+from lazyflow.operators.opReorderAxes import OpReorderAxes
+from lazyflow.operators.opUnblockedArrayCache import OpUnblockedArrayCache
+from lazyflow.request.request import Request, RequestLock, RequestPool
+from lazyflow.roi import getIntersectingRois, roiToSlice
+from lazyflow.rtype import SubRegion
+from lazyflow.slot import Slot
+from lazyflow.utility.helpers import get_ram_per_element
 
 logger = logging.getLogger(__name__)
 
+RoiTuple = Tuple[Tuple[int, ...], Tuple[int, ...]]
+Roi5D = Tuple[Tuple[int, int, int, int, int], Tuple[int, int, int, int, int]]
+
 
 class OpRelabelConsecutive(Operator):
+    """
+    Operator for relabeling label images in a consecutive manner
+
+    This operator sets up the inputs for OpRelabelConsecutive5D:
+    * `BlockShape` is set up in to whole time slices, no spatial subblocks
+      for valid relabelings.
+    * `dtype` of the `Input` is checked (see `supportedDtypes`) and in case of
+      `uint16` changed to `uint32` to comply with vigra requirements. `uint16` is
+       what we have seen with cellpose data.
+    * `Input` is always reordered to `tzyxc`
+    """
+
     Input = InputSlot()
-    StartLabel = InputSlot(value=0)
-    Output = OutputSlot()
+    StartLabel = InputSlot(value=1)
+
+    CachedOutput = OutputSlot()
+    RelabelDict = OutputSlot()
+
+    CleanBlocks = OutputSlot()
+    CompressionEnabled = InputSlot(value=False)
+
+    SerializationInput = InputSlot(optional=True)
+    SerializationOutput = OutputSlot()
+
+    # total dummy slot, there because OpObjectExtraction expects this interface
+    BypassModeEnabled = InputSlot(value=False)
```

Comment:
I think at least with this particular operator we should also cache in headless mode. So will remove bypassmodeenabled if possible without adding a lot of special checks downstream.

Reply:
I think the semantic dependency tracking here was just too hard to understand in what sense "OpObjectExtraction expects this interface"; now I get it.

I think in this context I would keep it on `OpRelabelConsecutive` to document the dependency (and the fact that OpRelabelConsecutive is actually an "OpLabelVolume", even if they share no base class or naming), but remove it from `OpRelabelConsecutive5D` so that at least that op can focus on its job